### PR TITLE
build tiny container with Dockerfile.scratch

### PR DESCRIPTION
**Pull request by HelloGrayson**
Brings image to 6mb, from 600mb:

```
REPOSITORY             TAG                 IMAGE ID            CREATED                  VIRTUAL SIZE
breerly/hello-server   9a47d50f            04941fdb328d        Less than a second ago   5.831 MB
```

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -22,7 +22,9 @@ after_success:
     - docker login -e $DOCKER_EMAIL -u $DOCKER_USER -p $DOCKER_PASS
     - export REPO=breerly/hello-server
     - export TAG=`if [ "$TRAVIS_BRANCH" == "master" ]; then echo "latest"; else echo $TRAVIS_BRANCH; fi`
-    - docker build -f Dockerfile -t $REPO:$COMMIT .
+    - CGO_ENABLED=0 GOOS=linux go build -a -installsuffix cgo -o main .
+    - docker build -f Dockerfile.scratch -t $REPO:$COMMIT .
+    - docker images
     - docker tag $REPO:$COMMIT $REPO:$TAG
     - docker tag $REPO:$COMMIT $REPO:travis-$TRAVIS_BUILD_NUMBER
     - docker push $REPO
```
Comment (abhinav) [inline .travis.yml:30]: Looks like this should be in a script at this point?

Comment (HelloGrayson) [inline .travis.yml:30]: perhaps - the weird part is that the script would depend on the travis environment

Comment (HelloGrayson) [inline .travis.yml:30]: (i actually started in a script then i realized i couldn't run it outside of travis' env)

File: README.md [hunk 2]
```diff
@@ -1,12 +1,38 @@
 Hello Server [![Build Status](https://travis-ci.org/breerly/hello-server.svg?branch=master)](https://travis-ci.org/breerly/hello-server)
 ============
 
-A Programmable Hello World Server.
+A Programmable, 6mb, Hello World Docker Appliance.
```
Comment (abhinav) [inline README.md:4]: really nitty nit: Not really programmable. Customizable, yes. Not programmable.

File: .travis.yml (hunk 1 above)
Comment (abhinav) [inline .travis.yml:30]: I mean, it's okay to have a directory travis/ or .travis/ with travis-specific scripts.

**Comment by abhinav** [comment]
I left my nits but :+1:

File: README.md (hunk 2 above)
Comment (HelloGrayson) [inline README.md:4]: Uh, yeah sure :)

Comment (HelloGrayson) [inline README.md:4]: It's only programmable in that it's customizable :P

**Pull request merged**